### PR TITLE
Update to net202 SDK

### PR DESCRIPTION
**Pull request by citizenmatt**
This PR will update SDK references to build against Rider 2020.2

- [x] Update SDK reference
- [x] Update versions, changelog and plugin metadata
- [x] Fix simple API changes in ReSharper plugin
- [x] Regenerate the icons to use correct namespace for moved APIs
- [x] Move highlighter registration from assembly based attributes to class based attributes
- [x] Fix warnings from deprecated APIs in frontend

Note that line mark renderers seem to be broken. This needs further investigation, as it may be a platform issue.

### CODE REVIEW

File: resharper/resharper-unity/src/CSharp/Daemon/Stages/PerformanceCriticalCodeAnalysis/Highlightings/PerformanceHighlightingAttributeIds.cs [hunk 1]
```diff
@@ -1,6 +1,57 @@
...
+        Layer = HighlighterLayer.SYNTAX)]
+        // ReSharper doesn't currently support EffectType.LINE_MARKER, so we'll use SOLID_UNDERLINE on the method name
+        // instead. Make sure the range is updated in any usages when this is removed!
+#if RIDER
+    [RegisterHighlighter(PERFORMANCE_CRITICAL_METHOD_HIGHLIGHTER,
+        GroupId = UnityHighlightingAttributeIds.GROUP_ID,
+        BackgroundColor = "#ff7526",
+        DarkBackgroundColor = "#ff7526",
+        EffectType = EffectType.LINE_MARKER,
+        Layer = HighlighterLayer.CARET_ROW - 1,
```
Comment (kashmervil) [inline resharper/resharper-unity/src/CSharp/Daemon/Stages/PerformanceCriticalCodeAnalysis/Highlightings/PerformanceHighlightingAttributeIds.cs:46]: This layer is way too low, I'd bump it to warning

File: resharper/resharper-unity/src/CSharp/Daemon/Stages/PerformanceCriticalCodeAnalysis/Highlightings/PerformanceHighlightingAttributeIds.cs [hunk 2]
```diff
@@ -1,6 +1,57 @@
...
 {
-    // Note that all attribute IDs should start with "ReSharper Unity " to appear properly in both Rider and ReSharper
+    // Note that these are registered as part of the main "Unity" group, as they should appear in the main "Unity"
+    // Color Scheme page
+    [RegisterHighlighter(COSTLY_METHOD_INVOCATION,
+        GroupId = UnityHighlightingAttributeIds.GROUP_ID,
+        EffectType = EffectType.SOLID_UNDERLINE,
+        EffectColor = "#ff7526",
+        NotRecyclable = true,
+        Layer = HighlighterLayer.SYNTAX)]
```
Comment (kashmervil) [inline resharper/resharper-unity/src/CSharp/Daemon/Stages/PerformanceCriticalCodeAnalysis/Highlightings/PerformanceHighlightingAttributeIds.cs:13]: I'd also use ADDITIONAL_SYNTAX or even higher, and leave syntax to lexer-based highlighting (comments, literals, etc.)

File: rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/toolWindow/UnityToolWindowManager.kt [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
 import com.jetbrains.rider.model.rdUnityModel
 import com.jetbrains.rider.plugins.unity.UnityHost
 import com.jetbrains.rider.projectView.solution
-import com.jetbrains.rider.util.idea.getLogger
 
 class UnityToolWindowManager(project: Project) : ProtocolSubscribedProjectComponent(project) {
+
     companion object {
-        private val myLogger = getLogger<UnityToolWindowManager>()
+        private val myLogger = Logger.getInstance(UnityToolWindowManager::class.java)
```
Comment (kashmervil) [inline rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/toolWindow/UnityToolWindowManager.kt:14]: IJ convention is `val LOG = logger< UnityToolWindowManager>()`, we'll eventually migrate to that (rdclient `val logger` is polluting import completion for idea project), mention this here just to say that there's a function `logger` in platform util 😸

Comment (citizenmatt) [reply rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/toolWindow/UnityToolWindowManager.kt:14]: Cool, good to know. I'll migrate these in a second PR. I've made a note of a few things that need fixing that aren't directly related to upgrading the SDK.

**Comment by citizenmatt** [comment]
Line mark rendering is due to a change in the platform. Added a workaround for now.

**Pull request merged**